plugins.go

```diff
@@ -17,6 +17,23 @@ type Plugin struct {
 	Params map[string]interface{}
 }
 
+func NewPluginFromReference(item interface{}) (*Plugin, error) {
+	switch item.(type) {
```

Comment:
You can do an assignment in the switch to save from having to do subsequent type assertions in each `case`. ie. 
```
switch i := item.(type) {
    case T:
        // Here i has type T
    case S:
        // Here i has type S
}
```
Check out https://tour.golang.org/methods/16

Reply:
Yeah, that will clean up that block of code a lot!

Reply:
Awesome! TIL, thanks for the link :)